models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055_numerator.sql

```diff
@@ -0,0 +1,238 @@
+{{ config(
+     enabled = var('quality_measures_enabled',var('claims_enabled',var('clinical_enabled',var('tuva_marts_enabled',False)))) | as_bool
+   )
+}}
+
+with denominator as (
+
+    select
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+    from {{ ref('quality_measures__int_cbe0055__denominator') }}
+
+)
+
+, retina_test_code as (
+
+    select
+          code
+        , code_system
+        , concept_name
+    From {{ref('quality_measures__value_sets')}}
+    where lower(concept_name) in  (
+          'diabetic retinopathy'
+        , 'ophthalmological services'
+        , 'diabetic retinal eye exam met'
+        , 'retinal or dilated eye exam'
+    )
+
+)
+
+, procedures as (
+
+    select
+        patient_id
+      , procedure_date
+      , coalesce (
+              normalized_code_type
+            , case
+                when lower(source_code_type) = 'cpt' then 'hcpcs'
+                when lower(source_code_type) = 'snomed' then 'snomed-ct'
+                else lower(source_code_type)
+              end
+          ) as code_type
+        , coalesce (
+              normalized_code
+            , source_code
+          ) as code
+    from {{ ref('quality_measures__stg_core__procedure') }}
+
+)
+
+, qualifying_procedure as (
+
+    select 
+          procedures.patient_id
+        , procedures.procedure_date
+        , procedures.code_type
+        , procedures.code
+        , retina_test_code.concept_name
+    from procedures
+    inner join {{ ref('quality_measures__int_cbe0055__performance_period') }} pp
+    on procedure_date between
+        performance_period_begin and performance_period_end
+    inner join retina_test_code
+        on procedures.code = retina_test_code.code
+            and procedures.code_type = retina_test_code.code_system
+
+)
+
+, qualifying_patients_for_procedure as (
+    select 
+          patient_id
+        , procedure_date
+        , code_type
+        , code
+        , concept_name
+    from qualifying_procedure
+    where lower(concept_name) in ( 
+                        'diabetic retinal eye exam met' 
+                      , 'retinal or dilated eye exam'
+                      )
+
+)
+
+, qualifying_patients_with_ophthalm_consult as (
+
+    select 
+        qualifying_procedure.patient_id,
+        qualifying_procedure.procedure_date,
+        qualifying_procedure.code_type,
+        qualifying_procedure.code
+    from qualifying_procedure
+    inner join qualifying_patients_for_procedure
+        on qualifying_procedure.patient_id = qualifying_patients_for_procedure.patient_id
+        and lower(qualifying_procedure.concept_name) = 'ophthalmological services'
```

Comment:
Not needed once the concept "ophthalmological services" is moved to the denominator.